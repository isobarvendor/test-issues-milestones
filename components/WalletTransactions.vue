<template>
  <div class="wallet-coins">
    <SliderHeader>Coin Transactions</SliderHeader>
    <div class="wallet-desktop">
      <div class="wallet-swiper" v-for="(item, index) in toBeShown" :key="'reward'+index">
        <div class="description bg">
            <div class="name">05 Nov 2019</div>
            <div class="collected">
              <span>Manchester United</span>
              <span class="negative">-50 points</span>
            </div>
            <div class="collected">
              <span>Coins Collected</span>
              <span class="positive">+10 points</span>
            </div>
          </div>
      </div>
    </div>
    <a class="button center" @click="nextPage" :disabled="currentPage == totalPages">View all</a>
  </div>
</template>

<script>
import SliderHeader from "@/components/WalletSliderHeader";
export default {
  name: "WalletTransactions",
  components: {
    SliderHeader
  },
  data() {
    return {
      currentPage: 1
    };
  },
  props: {
    data: null,
  },
  computed:{
  	toBeShown() {
    	return this.data.exclusivePrizes.slice(0, this.currentPage * 3);
    },
    totalPages() {
    	return Math.ceil( this.data.exclusivePrizes.length / 3);
    }
  },
  methods:{
    nextPage(){
    	if(this.currentPage <  this.totalPages) this.currentPage++;
    },
    prevPage(){
    	this.currentPage = this.currentPage - 1 || 1;
    }
  },
  beforeMount() {},
};
</script>
<style lang="scss">
  .wallet-coins{
    margin-bottom: 40px;
    .wallet-desktop{
      display: flex;
      flex-wrap: wrap;
      .wallet-swiper{
        margin: 30px 20px;
        width: 29%;
        .bg{
          background: linear-gradient(0deg, black 0%, #282828 100%);
          min-height: 140px;
        }
      }
      .description{
        padding: 20px;
        border-radius: 20px 20px 20px 20px;
        font-size: 14px;
        .name{
          font-size: 14px;
          color: #585858;
        }
        .name:after{
          content: '';
          display: block;
          margin: .5em 0;
          height: 1px;
          background: #585858;

        }
        .collected{
          display: flex;
          justify-content: space-between;
          margin-top: 10px;
          .negative{
            color: #DB1F27;
          }
          .positive{
            color: #FFC529;
          }
        }
      }
    }
    @media only screen and (max-width: 1199px) {
      .wallet-desktop{
        .wallet-swiper{
          width: 100%; 
          .bg{
            background: none;
            min-height: 140px;
            padding: 10px;
          }   
        }
      }  
    }
  }
</style>