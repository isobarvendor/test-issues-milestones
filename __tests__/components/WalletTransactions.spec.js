
import { shallowMount } from '@vue/test-utils'

import WalletTransactions from "@/components/WalletTransactions";


const factory = () => {
  return shallowMount(WalletTransactions, {

    propsData: {
      data :{
        transactions:[{
          walletRollingTotal:10
        }]
      }
    }
  });
};

describe("WalletTransactions", () => {


  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
