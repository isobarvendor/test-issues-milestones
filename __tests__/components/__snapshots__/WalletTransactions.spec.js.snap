// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletTransactions renders properly 1`] = `
<div class="wallet-coins">
  <sliderheader-stub>Coin Transactions</sliderheader-stub>
  <div class="wallet-desktop">
    <div class="wallet-swiper">
      <div class="description bg">
        <div class="name">05 Nov 2019</div>
        <div class="collected"><span>Manchester United</span> <span class="negative">10 points</span></div>
      </div>
    </div>
  </div> <a disabled="disabled" class="button center">View all</a>
</div>
`;
